playground/src/App.tsx

```diff
@@ -16,6 +17,79 @@ async function getExample(example: string) {
     return [(await inputs).text(), (await masm).text()];
   }
 
+
+function tryParseJSONObject(jsonString: string) {
+  try {
+    const o = JSON.parse(jsonString);
+
+    /**
+    * Handle non-exception-throwing cases:
+    * Neither JSON.parse(false) or JSON.parse(1234) throw errors, hence the type-checking,
+    * but... JSON.parse(null) returns null, and typeof null === "object", 
+    * so we must check for that, too. Thankfully, null is falsey, so this suffices:
+    */ 
+    if (o && typeof o === "object") {
+
+      return o;
+
+    }
+  }
+  catch (e) {
+
+    toast.error('Miden VM Inputs need to be a valid JSON object.');
+    return false;
+
+  }
+}
+
+/**
+* We check the inputs and return true or false. We allow: 
+* - an empty input, and
+* - a valid JSON object containing stack_init or advice_tape (or both) 
+* iff the values are numbers
+*/ 
+
+
+function checkInputs(jsonString: string) {
+  if (jsonString === "") {
+    return true;
+  }
+  if (!jsonString.includes("stack_init") && !jsonString.includes("advice_tape")) {
+
+    toast.error('Miden VM Inputs can be empty or we need either a stack_init or an advice_tape.');
+    return false;
+
+  }
+
+  const o = tryParseJSONObject(jsonString);
+
+  if (o === false) {
+    return false;
+  }
```

Comment:
It looks like this if statement wouldn't be necessary if the contents of `tryParseJSONObject` were moved in here. Given that it's not a long function and it's not used anywhere else (unless I missed it), I would probably prefer to move that completely into here.

I think a few other things could probably be condensed a bit here too. For example, we're doing the checks to see if the string includes "stack_init" and "advice_tape" twice.

Reply:
Got it, refactored quite a bit.